r/tests/testthat/test-dplyr-summarize.R

```diff
@@ -879,3 +879,20 @@ test_that("summarize() handles group_by .drop", {
     )
   )
 })
+
+test_that("summarise() passes through type information for temporary columns", {
+  # applies to ifelse and case_when(), in which argument types are checked
+  # within a translated function (previously this failed because the appropriate
+  # schema was not available for n() > 1, mean(y), and mean(z))
+  compare_dplyr_binding(
+    .input %>%
+      group_by(x) %>%
+      summarise(r = ifelse(n() > 1, mean(y), mean(z))) %>%
```

Comment:
This is super-duper minor, but I wonder if we should use `if_else()` here since that is a bit more strict about the types of the `true` and `false` arguments. IIRC, our arrow `ifelse()` is just as strict (since it uses the exact same implementation), but the intention of the test + testing the types of those would be a little bit clear with it.